### PR TITLE
feat: ssz encode and decode for uint

### PR DESCRIPTION
**Pull request by loocapro**
## Motivation
Completing SSZ implementation also for Uint types: https://github.com/alloy-rs/core/pull/407

## Solution
Implementing Encode and Decode traits from https://github.com/sigp/ethereum_ssz

## PR Checklist

- [x] Added Tests
- [ ] Added Documentation
- [x] Updated the changelog

### CODE REVIEW

File: src/support/ssz.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
+        if bytes.len() > nbytes(BITS) {
+            return Err(DecodeError::InvalidByteLength {
+                len:      bytes.len(),
+                expected: nbytes(BITS),
+            });
+        }
+        let mut buf = [0u8; nbytes(BITS)]; // generic parameters may not be used in const operations
```
Comment (loocapro) [inline src/support/ssz.rs:41]: Having some issue on using a generic parameter as a constant here.

Any advice @prestwich or @DaniPopes ?

Comment (prestwich) [reply src/support/ssz.rs:41]: You should be able to do `Ok(Self::from_le_slice(bytes))`, no?

**Review by prestwich — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 2]
```diff
@@ -108,6 +109,7 @@ std = [
     "serde?/std",
     "valuable?/std",
     "zeroize?/std",
+    "ethereum_ssz",
```
Comment (DaniPopes) [inline Cargo.toml:112]: This is not right

**Review by DaniPopes — CHANGES_REQUESTED** [review]
this needs a separate explicit feature instead of the implicit one

File: src/support/ssz.rs [hunk 3]
```diff
@@ -32,15 +32,7 @@ impl<const BITS: usize, const LIMBS: usize> Decode for Uint<BITS, LIMBS> {
...
-        if bytes.len() > nbytes(BITS) {
-            return Err(DecodeError::InvalidByteLength {
-                len:      bytes.len(),
-                expected: nbytes(BITS),
-            });
-        }
-        let mut buf = [0u8; nbytes(BITS)]; // generic parameters may not be used in const operations
-        buf[..bytes.len()].copy_from_slice(bytes);
-        Ok(Self::from_le_bytes(buf))
+        Ok(Self::from_le_slice(bytes))
```
Comment (prestwich) [inline src/support/ssz.rs:35]: still want the length check with custom error, as `from_le_slice` will panic on overflows
```
        if bytes.len() > nbytes(BITS) {
            return Err(DecodeError::InvalidByteLength {
                len:      bytes.len(),
                expected: nbytes(BITS),
            });
        }```

**Comment by prestwich** [comment]
Looks like tiny-keccak is `CC0 1.0` which is odd. We'll need to explicitly add that license to the approved list in `deny.toml`

**Review by prestwich — APPROVED** [review]

**Comment by prestwich** [comment]
okay last thing I promise, please update CHANGELOG.md 🖖

**Pull request merged**